### PR TITLE
feat!(storage): implement resourceVersion

### LINKED ISSUE #659: feat(storage): Implement resourceVersion in GetList method
The `GetList` method needs proper handling of the list’s `resourceVersion`, since it currently always returns `0`. 

Grafana shows an example of using snowflake IDs for this in its apiserver storage code: https://github.com/intergral/grafana/blob/2f390b310c1eec9d9cb7ee8459090d5dc19d3c02/pkg/apiserver/storage/file/file.go#L92.

 A similar approach could also be applied to the `Get` method.

### PR DESCRIPTION
**Pull request by fabriziosestito**
## Description

⚠️ This is a breaking change

This pull request introduces a global, monotonically increasing resource version sequence for all Kubernetes resources stored in the database, replacing the previous per-object resource versioning logic. The change ensures that resource versions are unique and ordered cluster-wide, which is essential for correct watch and list semantics. The pull request also updates the test suite to cover new behaviors and edge cases.

The `GetList` implementation is inspired by [Grafana's apiserver file storage](https://github.com/intergral/grafana/blob/2f390b310c1eec9d9cb7ee8459090d5dc19d3c02/pkg/apiserver/storage/file/file.go#L537).

Since we don't need (and can't) support historical data with pgsql backend, the [`Exact` `ResourceVersionMatch` semantic](https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list) is ignored. This matches the behavior of Kubernetes versions prior to v1.19, before `ResourceVersionMatch` was introduced.

The `NotOlderThan` semantic is supported by passing a non-empty, non-zero resource version. Objects with an RV lower than the requested value are filtered out at the database level.

When an empty or zero resource version is passed, a new RV is generated from the sequence. This is similar to what Grafana does and differs from the etcd-based Kubernetes implementation, which uses the current resource version. However, our approach is semantically valid and avoids a race condition where events could be missed between reading the list and obtaining the RV.

Note on the bigint sequence: 

A signed 64-bit integer gives:
Max: 9,223,372,036,854,775,807

Even at 1 million writes per second, exhaustion would take ~292,000 years

Fix #659

### CODE REVIEW

File: internal/storage/store.go [hunk 1]
```diff
@@ -503,7 +561,12 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 		nextContinueToken = strconv.FormatInt(lastID, 10)
 	}
 
-	if err = s.Versioner().UpdateList(listObj, 1, nextContinueToken, nil); err != nil {
+	rv, err := s.GetCurrentResourceVersion(ctx)
+	if err != nil {
+		return storage.NewInternalError(err)
+	}
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply internal/storage/store.go]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/kubewarden/sbomscanner/pull/709#discussion_r2627825554)

**Review by flavio — APPROVED** [review]
LGTM once the conflict is addressed
